libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ir/IrSchema.kt

```diff
@@ -129,10 +129,17 @@ internal class IrCompositeType2(val name: String) : IrType2
 
 @Serializable
 internal data class IrArgumentDefinition(
+    val id: String,
     val name: String,
+    val propertyName: String,
     val isKey: Boolean,
     val isPagination: Boolean,
-)
+) {
+  companion object {
+    fun id(type: String, field: String, argument: String) = "$type.$field.$argument"
+    fun propertyName(fieldName: String, argumentName: String) = "__${fieldName}_${argumentName}"
```

Comment:
For later: we should come up with some rules about symbols. Are `__` to escape nameclashes or a signal that it is not meant to be called directly. It's not 100% clear at the moment